tensorflow-framework/src/main/java/org/tensorflow/framework/initializers/RandomUniform.java

```diff
@@ -76,24 +75,19 @@ public RandomUniform(Ops tf, double minval, double maxval, long seed) {
 
   /** {@inheritDoc} */
   @Override
-  public <T extends TType> Operand<T> call(Operand<TInt64> dims, Class<T> type) {
-    if (!TNumber.class.isAssignableFrom(type)) {
-      throw new IllegalArgumentException("Tensor type must be numeric: " + type.getSimpleName());
-    }
-    @SuppressWarnings("unchecked")
-    Class<TNumber> nType = (Class<TNumber>) type;
-    Operand<TNumber> distOp;
+  public <U extends TNumber> Operand<U> call(Operand<TInt64> dims, Class<U> type) {
```

Comment:
Similar to `RandomNormal` is there a reason we don't check that the `minVal` is less than the `maxVal` on construction?